javascript/index.js.flow

```diff
@@ -3111,4 +3141,1732 @@ declare module "telegram-typings" {
      */
     score: number,
   };
+
+  /**
+   * Use this method to receive incoming updates using long polling (wiki). 
+   * An Array of Update objects is returned.
+   * @see http://en.wikipedia.org/wiki/Push_technology#Long_polling
+   * @see https://core.telegram.org/bots/api#update
+   */
+  declare type getUpdates = {
+    /**
+     * Optional
+     */
+    offset: number,
+
+    /**
+     * Optional
+     */
+    limit: number,
+
+    /**
+     * Optional
+     */
+    timeout: number,
+
+    /**
+     * Optional
+     */
+    allowed_updates: string[],
+  };
+
+  /**
+   * Use this method to specify a url and receive incoming updates via an 
+   * outgoing webhook. Whenever there is an update for the bot, we will send 
+   * an HTTPS POST request to the specified url, containing a JSON-serialized 
+   * Update. In case of an unsuccessful request, we will give up after a 
+   * reasonable amount of attempts. Returns true.
+   * @see https://core.telegram.org/bots/api#update
+   */
+  declare type setWebhook = {
+    /**
+     * Yes
+     */
+    url: string,
+
+    /**
+     * Optional
+     */
+    certificate: InputFile,
+
+    /**
+     * Optional
+     */
+    max_connections: number,
+
+    /**
+     * Optional
+     */
+    allowed_updates: string[],
+  };
+
+  /**
+   * Use this method to send text messages. On success, the sent Message is returned.
+   * @see https://core.telegram.org/bots/api#message
+   */
+  declare type sendMessage = {
+    /**
+     * Yes
+     */
+    chat_id: IntegerOrString,
+
+    /**
+     * Yes
+     */
+    text: string,
+
+    /**
+     * Optional
+     */
+    parse_mode: string,
+
+    /**
+     * Optional
+     */
+    disable_web_page_preview: boolean,
+
+    /**
+     * Optional
+     */
+    disable_notification: boolean,
+
+    /**
+     * Optional
+     */
+    reply_to_message_id: number,
+
+    /**
+     * Optional
+     */
+    reply_markup: InlineKeyboardMarkupOrReplyKeyboardMarkupOrReplyKeyboardRemoveOrForceReply,
```

Comment:
And here we will want to have:

```js
reply_markup: InlineKeyboardMarkup | ReplyKeyboardMarkup | ReplyKeyboardRemove | ForceReply
```

As the description says "Optional", should this then be 

```js
reply_markup?: InlineKeyboardMarkup | ReplyKeyboardMarkup | ReplyKeyboardRemove | ForceReply
```

right?

Reply:
Yes